frontend/src/features/dashboard/components/PlannedProductionModal.tsx

```diff
@@ -0,0 +1,367 @@
+/*
+Copyright (c) 2024 Volkswagen AG
+Copyright (c) 2024 Contributors to the Eclipse Foundation
+
+See the NOTICE file(s) distributed with this work for additional
+information regarding copyright ownership.
+
+This program and the accompanying materials are made available under the
+terms of the Apache License, Version 2.0 which is available at
+https://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+License for the specific language governing permissions and limitations
+under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import { useEffect, useMemo, useState } from 'react';
+import { Input, PageSnackbar, PageSnackbarStack, Table } from '@catena-x/portal-shared-components';
+import { UNITS_OF_MEASUREMENT } from '@models/constants/uom';
+import { Production } from '@models/types/data/production';
+import { Autocomplete, Box, Button, Dialog, DialogTitle, Grid, Stack, Typography, capitalize } from '@mui/material';
+import { getUnitOfMeasurement, isValidOrderReference } from '@util/helpers';
+import { usePartners } from '@features/stock-view/hooks/usePartners';
+import { deleteProduction, postProductionRange } from '@services/productions-service';
+import { Notification } from '@models/types/data/notification';
+import { Close, Delete, Save } from '@mui/icons-material';
+import { DateTime } from '@components/ui/DateTime';
+
+const GridItem = ({ label, value }: { label: string; value: string }) => (
+    <Grid item xs={6}>
+        <Stack>
+            <Typography variant="caption1" fontWeight={500}>
+                {label}:
+            </Typography>
+            <Typography variant="body3" paddingLeft=".5rem">
+                {value}
+            </Typography>
+        </Stack>
+    </Grid>
+);
+
+const createProductionColumns = (handleDelete: (row: Production) => void) =>
+    [
+        {
+            field: 'estimatedTimeOfCompletion',
+            headerName: 'Completion Time',
+            headerAlign: 'center',
+            width: 150,
+            renderCell: (data: { row: Production }) => (
+                <Box display="flex" textAlign="center" alignItems="center" justifyContent="center" width="100%" height="100%">
+                    {new Date(data.row.estimatedTimeOfCompletion).toLocaleTimeString('de-DE')}
```

Comment:
I found one strange behavior when doing some freehand testing: 12AM is considered as previous day (would have expected 00:00 as start of the selected day)
Reproduce:
- Open Dashboard > select production > select material and customer
- Create production > 
  - select day + 2 > default value for time is "12:00 AM"
  - confirm adding the production
Result: Production is added to day +1 instead of day + 2